examples/coalton-json/src/representation.lisp

```diff
@@ -66,4 +77,37 @@
     (Json-String  String)
     (Json-Number  Double-Float)
     (Json-Array   (List Json))
-    (Json-Object  (String-Map Json))))
+    (Json-Object  (String-Map Json)))
+
+  (declare string-map->string (String-Map json -> String))
+  (define (string-map->string sm)
+    (match sm
+      ((SM-Alist alist)
+       (let ((traverse (fn (current acc)
+                         (match current
+                           ((nil) acc)
+                           ((cons x xs)
+                            (match x
+                              ((tuple k v)
+                               (traverse xs (cons 
+                                             (concat5 "\"" k "\"" ":" (json->string v))
+                                             acc)))))))))
+         (let ((pairs (traverse alist (make-list))))
+           (lisp string (pairs)
+             (cl:format cl:nil "{~{~a~^, ~}}" pairs)))))))
+
+  (declare json->string (Json -> String))
+  (define (json->string jsn)
+    (match jsn
+      ((Json-Null) "nil")
+      ((Json-Boolean b) (if b "true" "false"))
+      ((Json-Number n) 
+       (lisp String (n)
+         (cl:format cl:nil "~$" n)))
```

Comment:
I'm not sure `~$` is the right float formatter here. In fact, I'm reasonably sure it is not; `~$` (with no additional arguments) prints exactly two digits after the decimal. This means that `(json->string (Json-Number 0.001))` will return `"0.00"`. I believe you want `~f`.

Reply:
updated